### PR TITLE
(WIP) Feature/documentation

### PR DESCRIPTION
**Pull request by joe94**
- [x] Place tr1d1um.yaml config file along with binary so users can run standalone `tr1d1um` right away 
- [x] Fix path to Dockerfile on Makefile
- [x] Add comments to tr1d1um.yaml config file
- [x] Verify local tr1d1um.yaml config values for standalone build works
- [x] Avoid ambiguity and have only one tr1d1um.yaml file in the repo. Can we do it?
- [x] Add explicit config value to turn on/off webhooks
- [x] `make docker` is broken... Fix it.

### CODE REVIEW

File: Dockerfile.local [hunk 1]
```diff
@@ -1,21 +1,9 @@
...
-
 # the actual image
 FROM alpine:latest
 RUN apk --no-cache add ca-certificates
+EXPOSE 6100 6101 6102
 RUN mkdir -p /etc/tr1d1um
 VOLUME /etc/tr1d1um
 WORKDIR /root/
-COPY --from=builder /go/src/tr1d1um_linux_amd64 .
+# we assume tr1d1um_linux_amd64 was already created before invoking this
```
Comment (kcajmagic) [inline Dockerfile.local]: We still need to copy the image over. 

Best practices is to build the binary in one container then copy it over.

File: README.md [hunk 2]
```diff
@@ -1,24 +1,106 @@
...
+# Tr1d1um
 
 [![Build Status](https://travis-ci.org/Comcast/tr1d1um.svg?branch=master)](https://travis-ci.org/Comcast/tr1d1um) 
 [![codecov.io](http://codecov.io/github/Comcast/tr1d1um/coverage.svg?branch=master)](http://codecov.io/github/Comcast/tr1d1um?branch=master)
 [![Go Report Card](https://goreportcard.com/badge/github.com/Comcast/tr1d1um)](https://goreportcard.com/report/github.com/Comcast/tr1d1um)
 [![Apache V2 License](http://img.shields.io/badge/license-Apache%20V2-blue.svg)](https://github.com/Comcast/tr1d1um/blob/master/LICENSE)
 
-The Webpa micro-service that encode TR-181 requests.
+An implementation of the [WebPA API](https://swagger.webpa.comcast.net) which supports TR-181 data model requests and leverages the
+[XMiDT API]((https://github.com/Comcast/xmidt)) under the hood.
```
Comment (kcajmagic) [inline README.md]: please update links to reference `xmidt-org`

File: Makefile [hunk 3]
```diff
@@ -88,4 +88,4 @@ it:
 
 .PHONY: clean
 clean:
-	rm -rf ./$(APP) ./.ignore ./coverage.txt ./vendor ./src/vendor
+	rm -rf $(APP)_linux_amd64 ./.ignore ./coverage.txt ./vendor ./src/vendor
```
Comment (kcajmagic) [inline Makefile]: we also need to remove the regular binary

File: README.md [hunk 4]
```diff
@@ -1,24 +1,106 @@
...
 
-## Centos 6
 
-1. Import the public GPG key (replace `0.0.1-65` with the release you want)
 
+
+## Summary
+The WebPA API operations can be divided into the following categories:
+
+1) **Device Statistics**
```
Comment (kcajmagic) [inline README.md]:
```suggestion
### Device Statistics
```
Since you specified categories it should be header rather than a list

File: README.md [hunk 5]
```diff
@@ -1,24 +1,106 @@
...
+
+- `GET /device/{device_id}/stat`
+
+Fetch the statistics for a given device connected to the XMiDT cluster
+
+2) **CRUD operations against devices connected to the XMiDT cluster**
```
Comment (kcajmagic) [inline README.md]:
```suggestion
### CRUD operations against devices connected to the XMiDT cluster
```

File: README.md [hunk 6]
```diff
@@ -1,24 +1,106 @@
...
+
+- `[GET|PATCH] /device/{device_id}/config`
+
+- `[POST|PUT] /device/{device_id}/config/{table}`
+
+- `DELETE /device/{device_id}/config/{row}`
+
+Tr1d1um validates the incoming request, injects it into the payload of a SimpleRequestResponse [WRP](https://github.com/Comcast/wrp-c/wiki/Web-Routing-Protocol) message and and sends it to XMiDT which then routes it to the specified device.
+
+3) **Event listener registration**
```
Comment (kcajmagic) [inline README.md]:
```suggestion
### Event listener registration
```

File: README.md [hunk 7]
```diff
@@ -1,24 +1,106 @@
...
+- `[POST/GET] /hook`
+
+## Details
```
Comment (kcajmagic) [inline README.md]: The current summary should be moved down to `details` and then we should have a 2-3 sentence summary of what Tr1d1um does.

File: README.md [hunk 8]
```diff
@@ -1,24 +1,106 @@
...
+The docker image can be built either with the Makefile or by running a docker 
+command.  Either option requires first getting the source code.
+
+See [Makefile](#Makefile) on specifics of how to build the image that way.
+
+For running a command, either you can run `docker build` after getting all 
+dependencies, or make the command fetch the dependencies.  If you don't want to 
+get the dependencies, run the following command:
+```bash
+docker build -t tr1d1um:local -f deploy/Dockerfile.local .
```
Comment (kcajmagic) [inline README.md]:
```suggestion
docker build -t tr1d1um:local -f Dockerfile.local .
```

File: README.md [hunk 9]
```diff
@@ -1,24 +1,106 @@
...
 ```
-yum install https://github.com/Comcast/tr1d1um/releases/download/0.0.1-65/tr1d1um-0.0.1-65.el6.x86_64.rpm
+If you want to get the dependencies then build, run the following commands:
+```bash
+docker build -t tr1d1um:local -f deploy/Dockerfile .
```
Comment (kcajmagic) [inline README.md]:
```suggestion
docker build -t tr1d1um:local -f Dockerfile .
```

File: README.md [hunk 10]
```diff
@@ -1,24 +1,106 @@
...
+For either command, if you want the tag to be a version instead of `local`, 
+then replace `local` in the `docker build` command.
+
+### Kubernetes
+
+WIP. TODO: add info
+
+## Deploy
+
+If you'd like to stand up `Tr1d1um` and the XMiDT cluster on Docker for local testing, refer to the [deploy README](https://github.com/Comcast/xmidt/tree/master/deploy/README.md).
```
Comment (kcajmagic) [inline README.md]:
```suggestion
If you'd like to stand up `Tr1d1um` and the XMiDT cluster on Docker for local testing, refer to the [deploy README](https://github.com/xmidt-org/xmidt/tree/master/deploy/README.md).
```

File: src/tr1d1um/tr1d1um.yaml [hunk 11]
```diff
@@ -0,0 +1,226 @@
...
+  # json is a flag indicating whether JSON logging output should be used.
+  # (Optional) defaults to false
+  json: true
+
+
+##############################################################################
+# Webhooks Related configuration 
+##############################################################################
+
+# webhooksEnables indicates whether or not the webhooks server should be started
```
Comment (kcajmagic) [inline src/tr1d1um/tr1d1um.yaml]:
```suggestion
# webhooksEnabled indicates whether or not the webhooks server should be started
```

File: src/tr1d1um/tr1d1um.yaml [hunk 12]
```diff
@@ -0,0 +1,226 @@
...
+########################################
+#   Logging Related Configuration
+########################################
+
+# log configures the logging subsystem details
+log:
+  # file is the name of the most recent log file.  If set to "stdout" this
+  # will log to os.Stdout.
+  # (Optional) defaults to os.TempDir()
+  #file: "/var/log/svalinn/svalinn.log"
```
Comment (kcajmagic) [inline src/tr1d1um/tr1d1um.yaml]:
```suggestion
  #file: "/var/log/tr1d1um/tr1d1um.log"
```
or delete it

**Comment by joe94** [comment]
@kcajmagic, I made changes based on your review. Thanks! 😎

File: README.md (hunk 7 above)
Comment (kristinaspring) [reply README.md]: I agree with Jack; what is under Summary looks more like a part of Details.  Usually the summary is just a 2-4 sentence paragraph that explains at a high level what the service does.
https://github.com/xmidt-org/svalinn#summary

File: README.md [hunk 13]
```diff
@@ -1,24 +1,103 @@
...
+
+- `[POST|PUT] /device/{device_id}/config/{table}`
+
+- `DELETE /device/{device_id}/config/{row}`
+
+Tr1d1um validates the incoming request, injects it into the payload of a SimpleRequestResponse [WRP](https://github.com/xmidt-org/wrp-c/wiki/Web-Routing-Protocol) message and and sends it to XMiDT which then routes it to the specified device.
+
+### Event listener registration
+
+- `[POST/GET] /hook`
```
Comment (kristinaspring) [inline README.md]: Can you add more explanation for what this does?

**Comment by joe94** [comment]
@kristinaspring and @kcajmagic, thanks for your good comments/ideas! I made some changes to the summary and details sections. 
A couple of points some of which are based on our offline convo:

- I think it's unnecessary to place actual `swagger-style` endpoints in the readme. It should suffice to give the reader just enough info to know which endpoints the section refers to.

- If we were to put ourselves in the shoes of someone trying to make actual requests against the tr1d1um API, we'd definitely need more details than what this readme provides. 

I'm thinking I should add more info about the fields mentioned (but not described) here https://swagger.webpa.comcast.net/#!/webpa/post_hook

As for the /config endpoints, it seems like they assume a certain level of familiarity with the tr-181 data model in a way that we may not need additional walkthrough. Wdyt, @schmidtw?

File: README.md [hunk 14]
```diff
@@ -1,24 +1,87 @@
-# tr1d1um
+# Tr1d1um
 
 [![Build Status](https://travis-ci.org/Comcast/tr1d1um.svg?branch=master)](https://travis-ci.org/Comcast/tr1d1um)
```
Comment (kcajmagic) [inline README.md]:
```suggestion
[![Build Status](https://travis-ci.org/xmidt-org/tr1d1um.svg?branch=master)](https://travis-ci.org/xmidt-org/tr1d1um) 
```

Comment (joe94) [reply README.md]: @kcajmagic, I don't think we can make this change until we port over travis itself as `https://travis-ci.org/xmidt-org/tr1d1um` doesn't work. I'll ask @schmidtw about this. I think he has the travis admin permissions.

**Comment by joe94** [comment]
@kcajmagic I just removed the swagger page!

**Pull request merged**